example/lib/pages/home.dart

```diff
@@ -19,80 +19,7 @@ class _HomePageState extends State<HomePage> {
-
-    const seenIntroBoxKey = 'seenIntroBox(a)';
-    if (kIsWeb && Uri.base.host.trim() == 'demo.fleaflet.dev') {
-      SchedulerBinding.instance.addPostFrameCallback(
-        (_) async {
-          final prefs = await SharedPreferences.getInstance();
-          if (prefs.getBool(seenIntroBoxKey) ?? false) return;
-
-          if (!mounted) return;
-
-          final width = MediaQuery.of(context).size.width;
-          await showDialog<void>(
-            context: context,
-            builder: (context) => AlertDialog(
-              icon: UnconstrainedBox(
-                child: SizedBox.square(
-                  dimension: 64,
-                  child:
-                      Image.asset('assets/ProjectIcon.png', fit: BoxFit.fill),
-                ),
-              ),
-              title: const Text('flutter_map Live Web Demo'),
-              content: ConstrainedBox(
-                constraints: BoxConstraints(
-                  maxWidth: width < 750
-                      ? double.infinity
-                      : (width / (width < 1100 ? 1.5 : 2.5)),
-                ),
-                child: Column(
-                  mainAxisSize: MainAxisSize.min,
-                  children: [
-                    const Text(
-                      "This is built automatically off of the latest commits to 'master', so may not reflect the latest release available on pub.dev.\nThis is hosted on Firebase Hosting, meaning there's limited bandwidth to share between all users, so please keep loads to a minimum.",
-                      textAlign: TextAlign.center,
-                    ),
-                    Padding(
-                      padding:
-                          const EdgeInsets.only(right: 8, top: 16, bottom: 4),
-                      child: Align(
-                        alignment: Alignment.centerRight,
-                        child: Text(
-                          "This won't be shown again",
-                          style: TextStyle(
-                            color: Theme.of(context)
-                                .colorScheme
-                                .inverseSurface
-                                .withOpacity(0.5),
-                          ),
-                          textAlign: TextAlign.right,
-                        ),
-                      ),
-                    ),
-                  ],
-                ),
-              ),
-              actions: [
-                TextButton.icon(
-                  onPressed: () => Navigator.of(context).pop(),
-                  label: const Text('OK'),
-                  icon: const Icon(Icons.done),
-                ),
-              ],
-              contentPadding: const EdgeInsets.only(
-                left: 24,
-                top: 16,
-                bottom: 0,
-                right: 24,
-              ),
-            ),
-          );
-          await prefs.setBool(seenIntroBoxKey, true);
-        },
-      );
-    }
+    _displayWebDemoIntroDialog();
```

Comment:
I'd prefer to not commit refactorings to the example project in this pr.
For more details: I wanted to do some clean up to the example project anyways and could include this as well. Mostly smaller things like using `MediaQuery.sizeOf` and so on.
This refactoring would suit there well too. What do you think?

Reply:
But a change that we could sneak in here would be there @override lint in `_SimulateErrorImageProvider` since https://github.com/fleaflet/flutter_map/pull/1704 might take another day or two to merge. 😅

Reply:
Thanks for pointing that second one out. Removing `@override` is not the correct solution here - `load` has been removed and wouldn't be called from anywhere! I've migrated to `loadImage` in 136a5b8.

I've also removed the changes to example/lib/pages/home.dart in a96724f.